From d9c9b99ebdc490c8682537d3e7ce6935d64a7dca Mon Sep 17 00:00:00 2001
From: Dante Russo <drusso@codeaurora.org>
Date: Fri, 24 Mar 2017 13:46:02 -0700
Subject: [PATCH 1/2] msm8974: Support moving conf files to /vendor/etc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Moving all vendor specific conf files to /vendor/etc instead of
current /etc folder

Test: building succeeded and tested on sailfish.
Bug: 36513600
CRs-Fixed: 1099981
Change-Id: I4495d8527941959be0847c4722ea8b68ee6c87ee
Signed-off-by: D. Andrei Măceș <andrei@unlegacy-android.org>
---
 msm8974/loc_api/libloc_api_50001/loc_eng.cpp | 39 +++++++++++++++++++-
 1 file changed, 37 insertions(+), 2 deletions(-)

diff --git a/msm8974/loc_api/libloc_api_50001/loc_eng.cpp b/msm8974/loc_api/libloc_api_50001/loc_eng.cpp
index 3c687e8..9f39cdc 100644
--- a/msm8974/loc_api/libloc_api_50001/loc_eng.cpp
+++ b/msm8974/loc_api/libloc_api_50001/loc_eng.cpp
@@ -2885,6 +2885,38 @@ static int set_sched_policy(int tid, SchedPolicy policy)
 }
 #endif /* USE_GLIB */
 
+/*===========================================================================
+FUNCTION resolve_config_file_path
+
+DESCRIPTION
+   resolve the given config file path into a vaild full path considering that
+   it could be located in vendor partition.
+
+DEPENDENCIES
+   None
+
+RETURN VALUE
+   None
+
+SIDE EFFECTS
+   N/A
+
+===========================================================================*/
+static void resolve_config_file_path(const char* conf_file_name,
+                                     char* resolved_file_path) {
+    FILE *file;
+    if (conf_file_name[0] == '/') {
+        sprintf(resolved_file_path, "/vendor%s", conf_file_name);
+    } else {
+        sprintf(resolved_file_path, "/vendor/%s", conf_file_name);
+    }
+    if ((file = fopen(resolved_file_path, "r")) != NULL) {
+        fclose(file);
+        return;
+    }
+    strcpy(resolved_file_path, conf_file_name);
+}
+
 /*===========================================================================
 FUNCTION    loc_eng_read_config
 
@@ -2910,8 +2942,11 @@ int loc_eng_read_config(void)
       loc_default_parameters();
       // We only want to parse the conf file once. This is a good place to ensure that.
       // In fact one day the conf file should go into context.
-      UTIL_READ_CONF(GPS_CONF_FILE, gps_conf_table);
-      UTIL_READ_CONF(SAP_CONF_FILE, sap_conf_table);
+      char conf_file_name[256];
+      resolve_config_file_path(GPS_CONF_FILE, conf_file_name);
+      UTIL_READ_CONF(conf_file_name, gps_conf_table);
+      resolve_config_file_path(SAP_CONF_FILE, conf_file_name);
+      UTIL_READ_CONF(conf_file_name, sap_conf_table);
       configAlreadyRead = true;
     } else {
       LOC_LOGV("GPS Config file has already been read\n");
-- 
2.17.1

